experimental/adaptive-cards-2/Catering/Bots/CateringBot.cs

```diff
@@ -0,0 +1,265 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+using AdaptiveCards.Templating;
+using Newtonsoft.Json;
+using Catering.Cards;
+using Catering.Models;
+using System.Net;
+using AdaptiveCards;
+using Newtonsoft.Json.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using Microsoft.Bot.AdaptiveCards;
+using System.Linq;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Bot.Builder.Dialogs;
+
+namespace Catering
+{
+    // This bot will respond to the user's input with an Adaptive Card.
+    // Adaptive Cards are a way for developers to exchange card content
+    // in a common and consistent way. A simple open card format enables
+    // an ecosystem of shared tooling, seamless integration between apps,
+    // and native cross-platform performance on any device.
+    // For each user interaction, an instance of this class is created and the OnTurnAsync method is called.
+    // This is a Transient lifetime service. Transient lifetime services are created
+    // each time they're requested. For each Activity received, a new instance of this
+    // class is created. Objects that are expensive to construct, or have a lifetime
+    // beyond the single turn, should be carefully managed.
+
+    public class CateringBot<TDialog> : ActivityHandler where TDialog : Dialog
+    {
+        private const string WelcomeText = "Welcome to the Adaptive Cards 2.0 Bot. This bot will introduce you to Action.Execute in Adaptive Cards.";
+        private BotState _userState;
+        private CateringDb _cateringDb;
+        private readonly CateringRecognizer _cateringRecognizer;
+        private readonly Dialog _dialog;
+
+        public CateringBot(UserState userState, CateringDb cateringDb, CateringRecognizer cateringRecognizer, TDialog dialog)
+        {
+            _userState = userState;
+            _cateringDb = cateringDb;
+            _cateringRecognizer = cateringRecognizer;
+            _dialog = dialog;
+        }
+
+        public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await base.OnTurnAsync(turnContext, cancellationToken);
+
+            await _userState.SaveChangesAsync(turnContext, false, cancellationToken);
+        }
+
+        protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersAdded, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.ChannelId == "directline" || turnContext.Activity.ChannelId == "webchat")
```

Comment:
Can you swap out the magic strings for Channels.Directline and Channels.Webchat via Microsoft.Bot.Connector?

Reply:
Fixed